packages/lib-core/src/store/PluginStore.ts

```diff
@@ -227,11 +228,28 @@ export class PluginStore implements PluginConsumer, PluginManager {
    */
   private isExtensionInUse(extension: Extension) {
     return (
-      (extension.flags?.required?.every((f) => this.options.isFeatureFlagEnabled(f)) ?? true) &&
-      (extension.flags?.disallowed?.every((f) => !this.options.isFeatureFlagEnabled(f)) ?? true)
+      (extension.flags?.required?.every((f) => this.featureFlags[f]) ?? true) &&
+      (extension.flags?.disallowed?.every((f) => !this.featureFlags[f]) ?? true)
     );
   }
 
+  setFeatureFlag = (flagName: string) => (newFlagValue: boolean) => {
```

Comment:
I'd prefer an API that takes 2 params. The hook can memoize a setter function.
@vojtechszocs  wdyt?